src/membraneWebRTC.ts

```diff
@@ -1372,6 +1379,22 @@ export class MembraneWebRTC {
     };
   };
 
+  private onIceCandidateError = (event: Event) => {
+    this.callbacks.onConnectionError?.("Ice candidate error occurred");
+  };
+
+  private onConnectionStateChange = (event: Event) => {
+    if (this.connection?.connectionState === "failed") {
+      this.callbacks.onConnectionError?.("Connection failed");
+    }
+  }
+
+  private onIceConnectionStateChange = (event: Event) => {
+    if (this.connection?.iceConnectionState === "failed") {
+      this.callbacks.onConnectionError?.("Ice connection failed");
+    }
```

Comment:
I think I'd also add a warn about `disconnected` state. It is recoverable (may get back to `connected`) but for debugging it might be useful to see it in console.

```suggestion
    if (this.connection?.iceConnectionState === "disconnected") {
      console.warn("ICE connection: disconnected");
    }
    if (this.connection?.iceConnectionState === "failed") {
      this.callbacks.onConnectionError?.("Ice connection failed");
    }
```

EDIT: I guess a switch might be used instead of ifs above